src/bin/main.rs

```
@@ -298,78 +327,74 @@ fn main() {
-            for (idx, code_file_pattern) in code_file_patterns.iter().enumerate() {
-                let code_path = paths.code_paths.as_ref().map(|code_paths| &code_paths[idx]);
-                for code_file in PathUtil::list_files(code_file_pattern).unwrap() {
-                    let out_path = code_path.map_or(code_file.clone(), |code_path| {
-                        modify_path(&code_file, &code_path)
-                    });
-                    match track_copy_dest.entry(out_path.clone()) {
-                        Occupied(entry) => {
-                            eprintln!(
-                                "ERROR: Attempted to copy multiple code files to {}: from {} and {}",
-                                out_path.display(), entry.get().display(), code_file.display()
-                            );
-                            return;
-                        }
-                        Vacant(entry) => {
-                            entry.insert(code_file.clone());
-                        }
-                    }
-                    eprintln!(
-                        "Copying code file {} to {}",
-                        code_file.display(),
-                        out_path.display()
-                    );
-
-                    let mut file_path = code_dir.clone();
-                    file_path.push(out_path);
-
-                    fs::create_dir_all(file_path.parent().unwrap()).unwrap();
-                    match fs::copy(&code_file, &file_path) {
-                        Ok(_) => {}
-                        Err(err) => eprintln!(
-                            "ERROR: --> Error copying code file {}: {}",
-                            code_file.display(),
-                            err
-                        ),
-                    }
+    for (idx, file_pattern) in patterns.iter().enumerate() {
+        let code_path = path_mod.as_ref().map(|code_paths| &code_paths[idx]);
+        for code_file in PathUtil::list_files(file_pattern).unwrap() {
```

If you are, cleaning this up: I don't `PathUtil` is actually a useful thing. For example in this case, it forces collecting the glob matching results into vector and it swallows any errors encountered during globbing. I think just using `glob::glob` directly here would be simpler and more efficient. (I think ideally `PathUtil` would go away completely, but this sort of leads to the whole overhaul/rewrite discussion again...)

---

The error handling is indeed suboptimal here, and could definitely be improved. I will first clean it from unused functions, and maybe try to get rid of it.

---

Got rid of `PathUtil` and probably handle errors appropriately now. But see https://github.com/mlange-42/yarner/pull/31#discussion_r545816458.